src/scaffold/emitter.rs

```diff
@@ -244,18 +244,41 @@ mod tests {
 
     use crate::constants::INTERNAL_DEFAULT_SOL_VERSION;
     use crate::error::Result;
+    use crate::hir::combiner::Combiner;
     use crate::hir::translator::Translator;
+    use crate::hir::Hir;
     use crate::scaffold::emitter;
     use crate::scaffold::modifiers;
+    use crate::syntax::ast::Ast;
     use crate::syntax::parser::Parser;
     use crate::syntax::tokenizer::Tokenizer;
 
     fn scaffold_with_flags(text: &str, indent: usize, version: &str) -> Result<String> {
-        let tokens = Tokenizer::new().tokenize(&text)?;
-        let ast = Parser::new().parse(&text, &tokens)?;
+        let combiner = Combiner::new();
+        let trees = combiner.split(text);
+
+        let mut parser = Parser::new();
         let mut discoverer = modifiers::ModifierDiscoverer::new();
-        let modifiers = discoverer.discover(&ast);
-        let hir = Translator::new().translate(&ast, modifiers);
+        let translator = Translator::new();
+
+        let asts: Vec<Ast> = trees
+            .iter()
+            .map(|&tree| {
+                let tokens = Tokenizer::new().tokenize(tree).unwrap();
+                parser.parse(tree, &tokens).unwrap()
+            })
+            .collect();
+
+        let hirs: Vec<Hir> = asts
+            .iter()
+            .map(|ast| {
+                let modifiers = discoverer.discover(ast);
+                translator.translate(ast, modifiers)
+            })
+            .collect();
+
+        combiner.verify(&asts).unwrap();
+        let hir = combiner.combine(&hirs);
```

Comment:
So, for this one it would be the same as before, but the result would be different. We would create a second function that calls this one for each tree, collects the hirs, and emits. This means that this one would be almost identical to `translate` + `entrypoint` as above, but adding the emitter step.